src/marks/vector.d.ts

```diff
@@ -3,29 +3,78 @@ import type {Data, FrameAnchor, MarkOptions, RenderableMark} from "../mark.js";
 
 export type VectorShapeName = "arrow" | "spike";
 
+/**
+ * A custom vector shape implementation.
+ */
 export interface VectorShapeImplementation {
   draw(context: CanvasPath, length: number, radius: number): void;
 }
 
 export type VectorShapeSpec = VectorShapeName | VectorShapeImplementation;
 
 export interface VectorOptions extends MarkOptions {
+  /** A channel for the horizontal position. */
   x?: ChannelValueSpec;
+  /** A channel for the vertical position. */
   y?: ChannelValueSpec;
+  /**
+   * The radius channel (in pixels) describes the mark’s secondary dimension
+   * (typically the *arrow* head’s span, or the *spike*’s base).
+   */
   r?: ChannelValueSpec;
+  /**
+   * The length of the shape, typically associated with a linear *length* scale.
+   */
   length?: ChannelValueSpec;
+  /** Vector rotation, in degrees; with rotate = 0, the shape points up. */
   rotate?: ChannelValue;
+  /** Vector shape, as a name (*arrow* or *spike*), or as a custom implementation. */
   shape?: VectorShapeSpec;
+  /** Vector anchor; by default a vector is anchored at its base (*start*). */
   anchor?: "start" | "middle" | "end";
+  /**
+   * The frame anchor, for vectors without a position, or positioned on a single
+   * (*x* or *y*) dimension in a two-dimensional frame.
```

Comment:
Tempted to enumerate the frame anchor options here, but I know we’ll have to repeat it in a few places. Might at least be worth stating that it can be one of the four corners, four edges, or the middle, as the anchor point if x and y are not specified. And also that it effectively provides a constant default for x and y, meaning that you can use it conjunction with either x or y (but that if you’ve already specified both x and y there is no reason to specify a frameAnchor).